object_store/src/memory.rs

```diff
@@ -82,7 +80,7 @@ impl ObjectStoreApi for InMemory {
             .get(location)
             .cloned()
             .context(NoDataInMemory {
-                location: location.display(),
+                location: format!("{}", location),
```

Comment:
```suggestion
                location: location.to_string(),
```
Since location now implements `Display` I think you can use `location.to_string()` instead here

Reply:
Good catch. Using `to_string()` here is better.